### ISSUE TITLE
buildout 2/3: Wheel file mode installed via buildout != pip install

### ISSUE THREAD

**Issue by delijati** [buildout/buildout#604]
This happens on buildout 2/3:
A executable has not the right mode when installed via buildout. I also removed the cached egg but still the same result.

I assumed `zc.recipe.egg` is just unpacking the wheels and not changing the permission on file.

Successfully installed setuptools-58.4.0 zc.buildout-2.13.3
Successfully installed pip-22.0.4 setuptools-60.9.3 zc.buildout-3.0.0rc2

buildout:
```
[buildout]
show-picked-versions = true

parts = py

eggs = kaleido 

[py]
recipe = zc.recipe.egg
eggs = ${buildout:eggs}
interpreter = py-foo
```

file mode via `pip install`:
```
❯ stat -c "%a" env/lib/python3.8/site-packages/kaleido/executable/kaleido
755
```

file mode via `buildout`:
```
❯ stat -c "%a" /home/xxx/.buildout/eggs/cp38/kaleido-0.2.1-py3.8-linux-x86_64.egg/kaleido/executable/kaleido 
644
```

**Comment by delijati** [comment]
Ok i narrowed it down to setuptools. It seams it copies the files but without the mode |-(.

```
import setuptools.wheel

if __name__ == "__main__":
    whl = setuptools.wheel.Wheel("kaleido-0.2.1.post1-py2.py3-none-manylinux2014_armv7l.whl")
    whl.install_as_egg("./foo")
```

**Comment by gotcha** [comment]
Do I understand correctly that the issue sits in `install_as_egg` ?

**Comment by delijati** [comment]
Jup i'm already filling out an issue in setuptools repo

**Comment by delijati** [comment]
I also added a fix to setuptools https://github.com/pypa/setuptools/pull/3167

**Comment by delijati** [comment]
it was merged ... yay :)

**Comment by gotcha** [comment]
Even better, it is released in https://pypi.org/project/setuptools/62.0.0/

**Comment by icemac** [comment]
Currently I run `zc.buildout=3.0.0rc2` with `setuptools < 60`. Is this max version of `setuptools` no longer needed?

**Comment by mauritsvanrees** [comment]
Should still be needed, except when you use an env for using distutils from core.

**Comment by delijati** [comment]
Yay i got now a working buildout setup:
```
$ virtualenv env -p /usr/bin/python3.9
$ env/bin/pip install -U zc.buildout==3.0.0rc2 setuptools
$ SETUPTOOLS_USE_DISTUTILS=stdlib env/bin/buildout -nvvv
$ bin/tests # this now works :)
```
The only still annoying part is this `SETUPTOOLS_USE_DISTUTILS=stdlib`. If it needs to be set anyway can it not just be then the default?

**Comment by gotcha** [comment]
@delijati Can this issue be closed ?

> The only still annoying part is this SETUPTOOLS_USE_DISTUTILS=stdlib. If it needs to be set anyway can it not just be then the default?

I plan to fix this before doing a final  `3.0` release. It will be tracked in #600.
